src/server/protocol.ts

```diff
@@ -679,8 +679,8 @@ namespace ts.server.protocol {
 
     export type OrganizeImportsScope = GetCombinedCodeFixScope;
 
-    export interface OrganizeImportsRequestArgs {
-        scope: OrganizeImportsScope;
+    export interface OrganizeImportsRequestArgs extends OrganizeImportsScope {
+        allowDestructiveCodeActions?: boolean;

```

Comment:
To confirm, we expect the editor to want to make this decision per-request and that's why it's not a UserPreference?

Reply:
Also i would name this flag as `skipDestructiveCodeActions` so that you would need to check only for truthiness instead of defaults when not set..

Reply:
>To confirm, we expect the editor to want to make this decision per-request and that's why it's not a UserPreference?
Yes, the editor would need to specify the option based on whether the command was sent as a part of `compileOnSave` or similar.

Reply:
dacfd7e renames to `skipDestructiveCodeActions` and defaults it to falsy.